app/src/renderer/App.tsx

```diff
@@ -1,66 +1,44 @@
-import { Button } from "antd";
+import { Routes, Route, Navigate } from "react-router";
 
-function App() {
-  const dummyRequest = async function () {
-    console.log("sending dummy request");
-    const res = await window.electronAPI.request({
-      method: "GET",
-      path_query: "/json",
-      headers: {},
-      stream: false,
-    });
-    console.log("received dummy response");
-    console.log(res);
-  };
-
-  const dummyStreamRequest = async function () {
-    console.log("sending dummy stream request");
-    const res = await window.electronAPI.requestStream(
-      {
-        method: "GET",
-        path_query: "/html",
-        headers: {},
-        stream: true,
-      },
-      "/tmp/download",
-    );
-    console.log("received dummy stream response");
-    console.log(res);
-  };
+import InboxView from "./views/Inbox";
+import SignInView from "./views/SignIn";
+import { useAppSelector } from "./hooks";
+import { SessionStatus } from "./features/session/sessionSlice";
 
+function App() {
   return (
-    <div>
-      <div className="max-w-sm rounded overflow-hidden shadow-lg">
-        <div className="px-6 py-4">
-          <div className="font-bold text-xl mb-2">Hello world!</div>
-          <p className="text-gray-700 text-base">
-            Lorem ipsum dolor sit amet, consectetur adipisicing elit.
-            Voluptatibus quia, nulla! Maiores et perferendis eaque,
-            exercitationem praesentium nihil.
-          </p>
-        </div>
-        <div className="px-6 pt-4 pb-2">
-          <Button
-            type="primary"
-            onClick={() => dummyRequest()}
-            title="Dummy Request"
-            data-testid="dummy-button"
-          >
-            Dummy Request
-          </Button>
-
-          <Button
-            type="default"
-            onClick={() => dummyStreamRequest()}
-            title="Dummy Stream Request"
-            data-testid="dummy-stream-button"
-          >
-            Dummy Stream Request
-          </Button>
-        </div>
-      </div>
-    </div>
+    <Routes>
+      <Route
+        index
+        path="/"
+        element={
+          <ProtectedRoute>
+            <InboxView />
+          </ProtectedRoute>
+        }
+      />
+      <Route path="/sign-in" element={<SignInView />} />
+    </Routes>
   );
 }
 
+// Gate the routes that require authentication
+
+type ProtectedRouteProps = {
+  children: React.ReactNode;
+};
+
+const ProtectedRoute = ({ children }: ProtectedRouteProps) => {
+  const session = useAppSelector((state) => state.session);
+
+  if (
+    session.status !== SessionStatus.Auth &&
+    session.status !== SessionStatus.Offline
+  ) {
+    return <Navigate to="/sign-in" replace />;
+  }
+
+  return children;
```

Comment:
Nit: The inverse logic might be cleaner here, so that we check for the cases we want (or else fail closed) rather than for the ones we don't want (or else "fail" open):

```suggestion
  if (
    session.status == SessionStatus.Auth ||
    session.status == SessionStatus.Offline
  ) {
    return children;
  }
  
  return <Navigate to="/sign-in" replace />;
```